hail/src/main/scala/is/hail/services/batch_client/BatchClient.scala

```diff
@@ -0,0 +1,163 @@
+package is.hail.services.batch_client
+
+import java.util.Random
+
+import is.hail.utils._
+import is.hail.services.{DeployConfig, Tokens}
+import org.apache.commons.io.IOUtils
+import org.apache.http.{HttpEntity, HttpEntityEnclosingRequest}
+import org.apache.http.client.methods.{HttpDelete, HttpGet, HttpPatch, HttpPost, HttpUriRequest}
+import org.apache.http.entity.{ByteArrayEntity, ContentType, StringEntity}
+import org.apache.http.impl.client.{CloseableHttpClient, HttpClients}
+import org.apache.http.util.EntityUtils
+import org.json4s.JsonAST.JNull
+import org.json4s.{DefaultFormats, Formats, JObject, JValue}
+import org.json4s.jackson.JsonMethods
+
+class NoBodyException(message: String, cause: Throwable) extends Exception(message, cause) {
+  def this() = this(null, null)
+
+  def this(message: String) = this(message, null)
+}
+
+class ClientResponseException(statusCode: Int, message: String, cause: Throwable) extends Exception(message, cause) {
+  def this(statusCode: Int) = this(statusCode, null, null)
+
+  def this(statusCode: Int, message: String) = this(statusCode, message, null)
+}
+
+class BatchClient extends AutoCloseable {
+  private[this] val baseUrl = DeployConfig.get.baseUrl("batch")
+
+  private[this] val httpClient: CloseableHttpClient = HttpClients.createDefault();
+
+  private[this] def request(req: HttpUriRequest, body: HttpEntity = null): JValue = {
+    if (body != null)
+      req.asInstanceOf[HttpEntityEnclosingRequest].setEntity(body)
+    Tokens.get.addServiceAuthHeaders("batch", req)
+    using(httpClient.execute(req)) { resp =>
+      val statusCode = resp.getStatusLine.getStatusCode
+      if (statusCode < 200 || statusCode >= 300) {
+        val entity = resp.getEntity
+        val message =
+          if (entity != null)
+            EntityUtils.toString(entity)
+        else
+            null
```

Comment:
the formatting here is kind of bizarre. Maybe use two lines?
```
val message =
  if (..) ...
  else ...
```

Reply:
Th else was just mis-indented.  I don't like the 2-line format because the alternatives are indented differently.